pkg/sql/ttl/ttljob/ttljob_test.go

```diff
@@ -581,6 +582,29 @@ func TestRowLevelTTLJobRandomEntries(t *testing.T) {
 				fmt.Sprintf(`SELECT count(1) FROM %s WHERE crdb_internal_expiration >= now()`, createTableStmt.Table.Table()),
 			).Scan(&numRows)
 			require.Equal(t, tc.numNonExpiredRows, numRows)
+
+			rows := th.sqlDB.Query(t, `
+SELECT sys_j.status, sys_j.progress
+FROM crdb_internal.jobs AS crdb_j
+JOIN system.jobs as sys_j ON crdb_j.job_id = sys_j.id
+WHERE crdb_j.job_type = 'ROW LEVEL TTL'
+`)
+			jobCount := 0
+			for rows.Next() {
+				var status string
+				var progressBytes []byte
+				require.NoError(t, rows.Scan(&status, &progressBytes))
+
+				require.Equal(t, "succeeded", status)
+
+				var progress jobspb.Progress
+				require.NoError(t, progress.Unmarshal(progressBytes))
+
+				rowLevelTTLProgress := progress.UnwrapDetails().(jobspb.RowLevelTTLProgress)
+				require.Equal(t, int64(tc.numExpiredRows), rowLevelTTLProgress.RowCount)
```

Comment:
`rowLevelTTLProgress.RowCount` is always 0 here even though I see it being updated in the job when debugging. Is there a step I am missing in the job itself to make sure it is persisted?